server/text_generation_server/models/ct2_causal_lm.py

```diff
@@ -0,0 +1,405 @@
+            ct2_compute_type = "bfloat16"
+        elif self.ct2_device == "cpu" and dtype in [torch.float16, torch.bfloat16]:
+            # float16 is not available on CPU
+            # and int16 has no stable implementation
+            ct2_compute_type = "float32"
+        else:
+            # default, int8 quantization.
+
+            if "cuda" in self.ct2_device:
+                # int8 for int8 layers, float16 for non-quantized layers
+                ct2_compute_type = "int8_float16"
+            else:
+                # int8 for int8 layers, float32 for non-quantized layers
+                ct2_compute_type = "int8"
+
+        # Start CT2 - conversion
+        out_dir = (
+            Path(HUGGINGFACE_HUB_CACHE)
+            / f"ct2models-{model_id.replace('/','--')}--{ct2_compute_type}"
+        )
+
+        if not os.path.exists(out_dir / "model.bin"):
+            try:
+                converter = ctranslate2.converters.TransformersConverter(
+                    model_id,
+                    activation_scales=None,
+                    load_as_float16=ct2_compute_type != "bfloat16",
+                    revision=revision,
+                    low_cpu_mem_usage=True,
+                    trust_remote_code=trust_remote_code,
+                )
+                converter.convert(
+                    output_dir=out_dir,
+                    vmap=None,
+                    quantization=ct2_compute_type,
+                    force=True,
+                )
+            except Exception as ex:
+                raise ValueError(
+                    f"conversion with ctranslate2 for {model_id} failed : Error {ex}"
+                )
+        if not os.path.exists(out_dir / "model.bin"):
+            raise ValueError(
+                f"no ctranslate2 model for {model_id} found after conversion in {out_dir}"
+            )
+
+        # Start CT2
+        self.ct2_model = ctranslate2.Generator(
+            str(out_dir),
+            device=self.ct2_device,
+            compute_type=ct2_compute_type,
+            **ct2_generator_kwargs,
+        )
+
+        class DummyModel(torch.nn.Module):
+            def __init__(self, *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.config = AutoConfig.from_pretrained(
+                    model_id, revision=revision, trust_remote_code=trust_remote_code
+                )
+
+        model = DummyModel()
+
+        if tokenizer.pad_token_id is None:
+            if model.config.pad_token_id is not None:
+                tokenizer.pad_token_id = model.config.pad_token_id
+            elif model.config.eos_token_id is not None:
+                tokenizer.pad_token_id = model.config.eos_token_id
+            elif tokenizer.eos_token_id is not None:
+                tokenizer.pad_token_id = tokenizer.eos_token_id
+            else:
+                tokenizer.add_special_tokens({"pad_token": "[PAD]"})
+
+        super(CT2CausalLM, self).__init__(
```

Comment:
Why not just `super()`?

Reply:
Just happened as I copied them over - I'll adapt that!